waiter/src/waiter/scheduler/cook.clj

```diff
@@ -488,8 +488,9 @@
      :syncer (retrieve-syncer-state-fn)})
 
   (validate-service [_ service-id]
-    (let [{:strs [run-as-user]} (service-id->service-description-fn service-id)]
-      (authz/check-user authorizer run-as-user service-id))))
+    (let [{:strs [run-as-user image]} (service-id->service-description-fn service-id)]
+      (authz/check-user authorizer run-as-user service-id)
+      (when image (throw (ex-info "Image field is set. Images are not supported with Marathon scheduler" {:image image}))))))
```

Comment:
Copy-paste error: Marathon should be Cook here.

Reply:
Would it be better to make a `reject-image-field` helper (maybe in scheduler.clj), and simply call it from each of the schedulers that doesn't support images? The scheduler name can be pulled from the `:name` on the scheduler object, and then added to the `ex-info` object along with the `:image` value.